stardata/Form1.cs

```diff
@@ -0,0 +1,235 @@
+                       
+                    }
+
+                    // Output the result to a text field
+                    Commentary.Text += "\r2). File: '" + filePath + "':\r\n\n" + text + "\n" + output;
+                    Commentary.Text += "\n3). A,B,C:"+ "\n" + ABCshowLetters + ABCoutput;
+                    Commentary.Text += "\n\n4). X,Y,Z:" + "\n" + XYZshowLetters + XYZoutput;
+
+                    
+
+                }
+                WriteToCsvFile(xValues, yValues, zValues, RadiusValues, NameValues);
+
+            }
+        }
+
+        private void button3_Click(object sender, EventArgs e)
+        {
+            //Verufy the user wants to exit
+            DialogResult result = MessageBox.Show("Are you sure you wish to quit?", "Exit App", MessageBoxButtons.YesNo);
+            if (result == System.Windows.Forms.DialogResult.Yes) 
+            {
+                System.Windows.Forms.Application.Exit(); 
+            }
+        }
+
+        private void WriteToCsvFile(List<double> xValues, List<double> yValues, List<double> zValues, List<double> RadiusValues, List<string> NameValues)
+        {
+            // Create a dialog box to select a file
+            SaveFileDialog saveFileDialog = new SaveFileDialog();
+            saveFileDialog.Filter = "CSV Files (*.csv)|*.csv";
+            saveFileDialog.Title = "Directory:";
+            saveFileDialog.FileName = "blenderXYZ.csv"; // Имя файла по умолчанию
+
+            // Open dialog box to select a file
+            DialogResult result = saveFileDialog.ShowDialog();
+
+            // If a file is selected, write data to the file
+            if (result == DialogResult.OK)
+            {
+                // Get the path to the selected file
+                string filePath = saveFileDialog.FileName;
+
+                // Create a dictionary to store already added values
+                Dictionary<string, bool> uniqueValues = new Dictionary<string, bool>();
+
+                // Write data to the file
+                using (StreamWriter file = new StreamWriter(filePath))
+                {
+                    for (int i = 0; i < xValues.Count; i++)
+                    {
+                        // Format a string to write to the file
+                        string lineOutput = $"{xValues[i]} {yValues[i]} {zValues[i]} {RadiusValues[i]} {NameValues[i]}";
+
+                        // Check if the value has already been added to the file
+                        if (!uniqueValues.ContainsKey(lineOutput))
+                        {
+                            // Add the value to the file
+                            file.WriteLine(lineOutput);
+
+                            // Add the value to the dictionary of already added values
+                            uniqueValues.Add(lineOutput, true);
+                        }
+                    }
+                }
+
+                MessageBox.Show("Data Saved");
+            }

```

Comment:
Overall, you may also include comments that provide explanations and instructions, enhancing code understandability for future developers or maintainers.